crates/nargo/src/cli/verify_cmd.rs

```diff
@@ -81,30 +80,26 @@ pub fn verify_with_path<P: AsRef<Path>>(
 }
 
 pub(crate) fn verify_proof(
-    mut compiled_program: CompiledProgram,
+    compiled_program: CompiledProgram,
     public_inputs_map: InputMap,
     proof: &[u8],
     verification_key: Vec<u8>,
 ) -> Result<bool, CliError> {
     let public_abi = compiled_program.abi.public_abi();
     let public_inputs =
-        public_abi.encode_to_array(&public_inputs_map).map_err(|error| match error {
+        public_abi.encode(&public_inputs_map, false).map_err(|error| match error {
```

Comment:
Okay so we now have this as false, so that we encode the return values/output in the public inputs list, since these are needed by the verifier